tests/test_pipelines.py

```diff
@@ -1139,23 +1138,48 @@ def test_stable_diffusion_memory_chunking(self):
 
     @slow
     @unittest.skipIf(torch_device == "cpu", "Stable diffusion is supposed to run on GPU")
-    def test_stable_diffusion_img2img_pipeline(self):
-        ds = load_dataset(
-            "imagefolder",
-            data_files={
-                "input": [
-                    "https://huggingface.co/datasets/hf-internal-testing/diffusers-images/resolve/main"
-                    "/img2img/sketch-mountains-input.jpg"
-                ],
-                "output": [
-                    "https://huggingface.co/datasets/hf-internal-testing/diffusers-images/resolve/main"
-                    "/img2img/fantasy_landscape.png"
-                ],
-            },
+    def test_stable_diffusion_text2img_pipeline(self):
+        output_image = load_image(
+            "https://huggingface.co/datasets/hf-internal-testing/diffusers-images/resolve/main"
+            "/text2img/astronaut_riding_a_horse.png"
+        )
+
+        model_id = "CompVis/stable-diffusion-v1-4"
+        pipe = StableDiffusionPipeline.from_pretrained(
+            model_id,
+            use_auth_token=True,
         )
+        pipe.to(torch_device)
+        pipe.enable_attention_slicing()
+        pipe.set_progress_bar_config(disable=None)
+
+        prompt = "astronaut riding a horse"
+
+        generator = torch.Generator(device=torch_device).manual_seed(0)
+        with torch.autocast("cuda"):
+            output = pipe(prompt=prompt, strength=0.75, guidance_scale=7.5, generator=generator)
+        image = output.images[0]
+
+        expected_array = np.array(output_image, dtype=np.int32)
+        sampled_array = np.array(image, dtype=np.int32)
 
-        init_image = ds["input"]["image"][0].resize((768, 512))
-        output_image = ds["output"]["image"][0].resize((768, 512))
+        assert sampled_array.shape == (512, 512, 3)
+        # using the mean absolute error due to slightly inconsistent outputs across different GPUs
+        assert np.mean(np.abs(sampled_array - expected_array)) < 2.0
```

Comment:
2.0 seems to large to me here as discussed. Can we see how big the difference is for pure numpy arrays without **any** optimization? 

Will run:

```python
model_id = "CompVis/stable-diffusion-v1-4"
pipe = StableDiffusionPipeline.from_pretrained(
    model_id,
    use_auth_token=True,
)
pipe.to(torch_device)

prompt = "A fantasy landscape, trending on artstation"
generator = torch.Generator(device=torch_device).manual_seed(0)
output = pipe(prompt=prompt, guidance_scale=7.5, generator=generator, output_type="np")
```

from current master @anton-l could you do the same and then we compare numpy results across `vorace`, `colab` and `brutasse`

Reply:
Note that I just merged @patil-suraj improved attention slicing when disabled, so let's try to include this PR when running the tests